modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/SabrSwaptionRawDataSensitivityCalculator.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.swaption;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import com.google.common.collect.ImmutableList;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.data.MarketDataName;
+import com.opengamma.strata.market.param.CurrencyParameterSensitivities;
+import com.opengamma.strata.market.param.CurrencyParameterSensitivity;
+import com.opengamma.strata.market.param.ParameterMetadata;
+import com.opengamma.strata.market.surface.SurfaceName;
+
+/**
+ * Calculator to obtain the raw data sensitivities for swaption related products using calibrated SABR data.
+ * <p>
+ * This needs data sensitivity info obtained during curve calibration.
+ */
+public class SabrSwaptionRawDataSensitivityCalculator {
+
+  /**
+   * The default instance.
+   */
+  public static final SabrSwaptionRawDataSensitivityCalculator DEFAULT = new SabrSwaptionRawDataSensitivityCalculator();
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the raw data sensitivities from SABR parameter sensitivity.
+   * <p>
+   * The SABR parameter sensitivities to data are stored in some optional data in the 
+   * {@link SabrParametersSwaptionVolatilities}.
+   * The sensitivities to the SABR parameters passed in should be compatible with the SABR parameters in term of data order.
+   * <p>
+   * Only the sensitivity to the SABR parameters for which there is a data sensitivity are taken into account. 
+   * At least one of the four parameter must have such sensitivities.
+   * 
+   * @param paramSensitivities  the curve SABR parameter sensitivities
+   * @param volatilities  the SABR parameters, including the data sensitivity metadata
+   * @return the raw data sensitivities
+   */
+  public CurrencyParameterSensitivity parallelSensitivity(
+      CurrencyParameterSensitivities paramSensitivities,
+      SabrParametersSwaptionVolatilities volatilities) {
+    boolean[] dataSensitivityInfoAvailable = new boolean[4]; // parameters in Alpha, Beta, Rho, Nu order
+    List<List<DoubleArray>> sensitivityToRawData = new ArrayList<>(4);
+    Optional<ImmutableList<DoubleArray>> alphaInfo = volatilities.getDataSensitivityAlpha();
+    dataSensitivityInfoAvailable[0] = alphaInfo.isPresent();
```

Comment:
Clearer to use alphaInfo.isPresent() elsewhere and delete `dataSensitivityInfoAvailable[]`.